Sources/Widgets/Widgets3D/SplineWidget/behavior.js

```diff
@@ -273,8 +273,8 @@ export default function widgetBehavior(publicAPI, model) {
       return macro.VOID;
     }
 
-    model.manipulator.setOrigin(model.activeState.getOrigin());
     model.manipulator.setNormal(model.camera.getDirectionOfProjection());
+
```

Comment:
why?

Reply:
This is related to the manipulator discussion. The widget works fine (as far as i can tell) without this line and from the discussions i had with @finetjul , new flavored vtk.js widgets should have an update to describe correctly the context in which they are used (e.g imageSlice, 3D scene ...)

Reply:
ok my bad i didn't read #1422 .

Reply:
This line plants the widget on the paint example. But when i add it after activeState is set with worldCoords (see [here](https://github.com/Kitware/vtk-js/blob/36575579ca1650c41d8575c830a7ba5edbce7bab/Sources/Widgets/Widgets3D/SplineWidget/behavior.js#L302)), all works fine. My guess is that activeState is not set properly before the 1st call to handleEvent

Reply:
This line plants the whole widget. I'm not sure why yet, but it seems that setting the plane origin with the activeState as it was uses the value of the state (see [here](https://github.com/Kitware/vtk-js/blob/1d65f61b066a00fdcb2e7da1add8f4c523f2f6ee/Sources/Widgets/Widgets3D/SplineWidget/state.js#L18) to compute the 1st position, which messes all up since it would then use corrupted data

Reply:
@floryst can you share your thoughts on this specifically?
```js
model.manipulator.setOrigin(model.activeState.getOrigin())
```

Reply:
ping @floryst. Also done here: https://github.com/Kitware/vtk-js/pull/1717/files#diff-09c501501761db156d6f7d2d7b93d812c1fccc886aac5a921dbda49b32481362L387-L388

Thanks!

Reply:
I think this should never happen once we have a discussion and implementation on updating manipulators based on context. I don't know how this works in the context of this widget, but usually state is updated based on the current manipulator location, not the other way around. If the activeState.getOrigin() is correct, then this is fine for now, but I would want to move away from this approach of setting manipulator parameters.

Reply:
@agirault do you think the PR is good for merge then?

Reply:
See other comments @ocrossi, but that specific issue is good: indeed, the state should not set the manipulator, but the contrary should occur. i.e. Manipulator setup + interaction = 2d screen coordinates to 3d world coordinates conversion for all the inner states of the widget.